### PR TITLE
[BEAM-4135] Stop taking the whole result in WatermarkManager

### PR DESCRIPTION
**Pull request by tgroh**
Instead, take in the exploded components which are relevant to the WatermarkManager.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand:
   - [ ] What the pull request does
   - [ ] Why it does it
   - [ ] How it does it
   - [ ] Why this approach
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by tgroh** [comment]
run java precommit

**Comment by tgroh** [comment]
R: @huygaa11

File: runners/direct-java/src/main/java/org/apache/beam/runners/direct/WatermarkManager.java [hunk 1]
```diff
@@ -991,11 +997,13 @@ private void applyPendingUpdate(PendingWatermarkUpdate<ExecutableT> pending) {
   private void updatePending(
```
Comment (huygaa11) [inline runners/direct-java/src/main/java/org/apache/beam/runners/direct/WatermarkManager.java]: Add "@param"'s

File: runners/direct-java/src/main/java/org/apache/beam/runners/direct/WatermarkManager.java [hunk 2]
```diff
@@ -965,15 +967,19 @@ private void applyNUpdates(int numUpdates) {
   }
 
   private void applyPendingUpdate(PendingWatermarkUpdate<ExecutableT> pending) {
-    CommittedResult<ExecutableT> result = pending.getResult();
-    ExecutableT transform = result.getExecutable();
+    ExecutableT transform = pending.getExecutable();
```
Comment (huygaa11) [inline runners/direct-java/src/main/java/org/apache/beam/runners/direct/WatermarkManager.java]: Are transform and executable are interchangeable term? Or, is transform a subset of executable?

File: runners/direct-java/src/main/java/org/apache/beam/runners/direct/WatermarkManager.java [hunk 3]
```diff
@@ -1542,18 +1539,22 @@ public int compare(CommittedBundle<?> o1, CommittedBundle<?> o2) {
 
     abstract TimerUpdate getTimerUpdate();
 
-    abstract CommittedResult<ExecutableT> getResult();
+    @Nullable
```
Comment (huygaa11) [inline runners/direct-java/src/main/java/org/apache/beam/runners/direct/WatermarkManager.java]: Could you copy paste the @param descriptions from other methods to here?

Comment (huygaa11) [reply runners/direct-java/src/main/java/org/apache/beam/runners/direct/WatermarkManager.java]: sounds good

**Pull request merged**